### PR TITLE
Make Remotery use vanilla hashfn again

### PR DESCRIPTION
**Pull request by JCash**

### CODE REVIEW

File: engine/dlib/src/remotery/lib/Remotery.c [hunk 1]
```diff
@@ -66,10 +66,8 @@
 
 // DEFOLD
 #include <dmsdk/dlib/log.h>
-#include <dmsdk/dlib/hash.h>
-
 static const char* g_EmptyString = "<empty>"; // As seen in profile_remotery.cpp _rmt_HashString32()
-static rmtU32 g_EmptyHash = 0;
+static rmtU32 g_EmptyHash = 0x89abcdef; // Unlikely anything else would collide with this
```
Comment (JCash) [inline engine/dlib/src/remotery/lib/Remotery.c:70]: Could technically call `_rmt_HashString32()` on the string, but this avoids doing it at global scope.

File: engine/profiler/src/profiler.cpp [hunk 2]
```diff
@@ -541,7 +542,8 @@ static void ProcessSample(dmProfileRender::ProfilerThread* thread, int indent, d
     out.m_Count = dmProfile::SampleGetCallCount(sample);
     out.m_Color = dmProfile::SampleGetColor(sample);
     out.m_Indent = (uint8_t)indent;
-    out.m_NameHash = dmProfile::SampleGetNameHash(sample);
+    const char* name = dmProfile::SampleGetName(sample);
+    out.m_NameHash = dmHashString32(name?name:"<empty_sample_name>");
```
Comment (JCash) [inline engine/profiler/src/profiler.cpp:546]: Now we need to hash this every frame instead.

File: engine/profiler/src/profiler.cpp [hunk 3]
```diff
@@ -594,7 +596,9 @@ static void SampleTreeCallback(void* _ctx, const char* thread_name, dmProfile::H
 
 static void ProcessProperty(dmProfileRender::ProfilerFrame* frame, int indent, dmProfile::HProperty property)
 {
-    uint32_t name_hash = dmProfile::PropertyGetNameHash(property);
+    const char* name = dmProfile::PropertyGetName(property);
+    uint32_t name_hash = dmHashString32(name?name:"<empty_property_name>");
+
```
Comment (JCash) [inline engine/profiler/src/profiler.cpp:601]: Now we need to hash this every frame instead.

Comment (AGulev) [reply engine/profiler/src/profiler.cpp:601]: ;(

**Review by AGulev — APPROVED** [review]
I think we should try it

**Review by britzl — APPROVED** [review]

**Pull request merged**